### PR TITLE
Update remove fields

### PR DESCRIPTION
**Pull request by tpfau**
Added a sanity check to catch models with invalid rules fields, as they would otherwise get stuck in the while loop.

**I hereby confirm that I have:**

- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [X] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [] with [X] to check the box)*

### CODE REVIEW

File: src/base/io/readCbModel.m [hunk 1]
```diff
@@ -353,10 +353,11 @@
                     end
                 end
             end
-        catch
+        catch ME
             % IF we are here, there was a problem in verifyModel or convertOldStyleModel, so this is
             % not a model.
-    end
+            0;
```
Comment (laurentheirendt) [inline src/base/io/readCbModel.m]: Why is there a `0`?

File: src/reconstruction/refinement/removeFieldEntriesForType.m [hunk 2]
```diff
@@ -76,7 +76,11 @@
     end
     %update the rules fields.
     if isfield(model,'rules') %Rely on rules first  
-        %However, we first normalize the rules.
+        rulesFieldOk = verifyModel(model,'simpleCheck',true,'restrictToFields',{'rules'}, 'silentCheck', true);
+        if ~rulesFieldOk
+            error('Rules field does not satisfy the field definitions. Please check that it satisfies the definitions given <a href="https://github.com/opencobra/cobratoolbox/blob/master/docs/source/notes/COBRAModelFields.md">here</a>');
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/removeFieldEntriesForType.m]: May you implement the error message using [hyperlink.m](https://github.com/opencobra/cobratoolbox/blob/master/src/base/hyperlink.m) such that the error message is correctly displayed when MATLAB is executed in console mode?

Comment (tpfau) [reply src/reconstruction/refinement/removeFieldEntriesForType.m]: Done.

File: src/base/io/readCbModel.m (hunk 1 above)
Comment (tpfau) [reply src/base/io/readCbModel.m]: Debugging... 
Needed a breakpoint, so I needed some code.
Removed it.

**Comment by laurentheirendt** [comment]
cc: @jennmodamio

**Pull request merged**